mobile/components/Icon/index.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import * as React from 'react';
+import TouchableOpacity from '../TouchableOpacity';
+import ConditionalWrap from '../ConditionalWrap';
+import { Svg } from 'expo';
+const { G } = Svg;
+import { withTheme } from 'styled-components/native';
+import type { GlyphTypes } from './types';
+import getPaths from './glyphs';
+
+type Props = {
+  onPress?: Function,
+  glyph: GlyphTypes,
+  size?: number,
+  color?: string,
+  theme: Object,
+};
+
+// prettier-ignore
+export default withTheme(({ size = 64, color = 'text.alt', glyph, theme, onPress }: Props) => {
+    const colorPaths = color.split('.')
+    const fill = theme[colorPaths[0]][colorPaths[1]]
```

Comment:
This is pretty hacky and brittle, let's make this a function instead which gets passed the theme similar to what you'd get in a styled component directly:

```JS
type Props = {
 color?: (theme: Object) => string,
};

const fill = color ? color(theme) : theme.text.alt;
```